### PR TITLE
feature: support version valid check

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

feature: support version valid check

### CODE REVIEW

File: core/src/main/java/io/seata/core/protocol/Version.java [hunk 1]
```diff
@@ -69,8 +76,35 @@ public static String getChannelVersion(Channel c) {
...
      * @throws IncompatibleVersionException the incompatible version exception
      */
-    public static String checkVersion(String version) throws IncompatibleVersionException {
-        // TODO: check
-        return version;
+    public static void checkVersion(String version) throws IncompatibleVersionException {
+        long current = convertVersion(CURRENT);
+        long clientVersion = convertVersion(version);
+        long divideVersion = convertVersion(VERSION_0_7_1);
+        if ((current - divideVersion > 0 && clientVersion - divideVersion < 0) || (current - divideVersion < 0 && clientVersion - divideVersion > 0)) {
```
Comment (ph3636) [inline core/src/main/java/io/seata/core/protocol/Version.java:83]: Is it clearer to change 【current - divideVersion > 0】 to 【current > divideVersion】

File: core/src/main/java/io/seata/core/protocol/Version.java [hunk 2]
```diff
@@ -69,8 +76,35 @@ public static String getChannelVersion(Channel c) {
...
+            throw new IncompatibleVersionException("incompatible client version:" + version);
+        }
+    }
+
+    private static long convertVersion(String version) throws IncompatibleVersionException {
+        String parts[] = StringUtils.split(version, '.');
+        Long result = 0L;
+        int i = 1;
+        int size = parts.length;
+        if (size > MAX_VERSION_DOT + 1) {
```
Comment (ph3636) [inline core/src/main/java/io/seata/core/protocol/Version.java:93]: Why not fix it to 3?  
if (size != 3 ) {
throw ...
}

Comment (slievrly) [reply core/src/main/java/io/seata/core/protocol/Version.java:93]: some version such as： 1.3.0，1.3.0.1，1.3.0-SNAPSHOT ...

File: core/src/main/java/io/seata/core/protocol/Version.java (hunk 1 above)
Comment (slievrly) [reply core/src/main/java/io/seata/core/protocol/Version.java:83]: fixed

File: core/src/main/java/io/seata/core/protocol/Version.java (hunk 2 above)
Comment (ph3636) [reply core/src/main/java/io/seata/core/protocol/Version.java:93]: If so, The value of result is that 1.3.0.1 is larger than 1.4.0?

Comment (slievrly) [reply core/src/main/java/io/seata/core/protocol/Version.java:93]: get what you mean. fix later

Comment (slievrly) [reply core/src/main/java/io/seata/core/protocol/Version.java:93]: @ph3636 how about this one?

Comment (ph3636) [reply core/src/main/java/io/seata/core/protocol/Version.java:93]: good

File: core/src/main/java/io/seata/core/protocol/Version.java [hunk 3]
```diff
@@ -69,8 +76,36 @@ public static String getChannelVersion(Channel c) {
...
      * @throws IncompatibleVersionException the incompatible version exception
      */
-    public static String checkVersion(String version) throws IncompatibleVersionException {
-        // TODO: check
-        return version;
+    public static void checkVersion(String version) throws IncompatibleVersionException {
+        long current = convertVersion(CURRENT);
+        long clientVersion = convertVersion(version);
+        long divideVersion = convertVersion(VERSION_0_7_1);
+        if ((current > divideVersion && clientVersion < divideVersion) || (current < divideVersion && clientVersion > divideVersion)) {
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/protocol/Version.java:83]: The verification function has not yet been reached and an exception has been thrown: io.netty.handler.codec.TooLongFrameException: Adjusted frame length exceeds 8388608: 2415945472-discarded

**Pull request merged**